lib/std/rand.zig

```diff
@@ -47,6 +47,30 @@ pub const Random = struct {
         return r.int(u1) != 0;
     }
 
+    /// Returns a random value from an enum, evenly distributed.
+    pub fn enumValue(r: *Random, comptime EnumType: type) EnumType {
+        if (comptime !std.meta.trait.is(.Enum)(EnumType)) {
+            @compileError("Random.enumValue requires an enum type, not a "++@typeName(EnumType));
+        }
+
+        // We won't use int -> enum casting because enum elements can have
+        //  arbitrary values.  Instead we'll randomly pick one of the type's
+        //  fields (values).
+        const options = comptime std.meta.fields(EnumType);
+        if (options.len == 0) {
+            @compileError("Cannot get a random value from an empty enum");
+        }
+        const index = r.uintLessThan(u64, options.len);
+        inline for (options) |field, i| {
+            if (i == index) {
+                return @field(EnumType, field.name);
+            }
+        }
+
+        // The above loop should be exhaustive
```

Comment:
The general pattern is to put this comment after the unreachable on the same line:
```zig
unreachable; // The above loop should be exhaustive
```
so that the stack traces show it.